kroxylicious-runtime/src/main/templates/BodyDecoder.ftl

```diff
@@ -65,17 +64,34 @@ public class BodyDecoder {
     * @return the ApiMessage
     * @throws IllegalArgumentException if an unhandled ApiKey is encountered
     */
-    static ApiMessage decodeResponse(ApiKeys apiKey, short apiVersion, Readable accessor) {
-        switch (apiKey) {
+    static ApiMessage decodeResponse(ApiKeys apiKey, short apiVersion, ByteBufAccessor accessor) {
+        return switch (apiKey) {
 <#list messageSpecs as messageSpec>
     <#if messageSpec.type?lower_case == 'response'>
-            case ${retrieveApiKey(messageSpec)}:
-                return new ${messageSpec.name}Data(accessor, apiVersion);
+        <#if messageSpec.name == 'ApiVersionsResponse'>
+            case ${retrieveApiKey(messageSpec)} -> {
```

Comment:
I'd suggest adding a comment here to explain that the fallback algorithm is from Kafka and link to https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/requests/ApiVersionsResponse.java#L90-L106